sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/routing/LocationCache.java

```diff
@@ -207,40 +213,50 @@ public UnmodifiableList<URI> getApplicableWriteEndpoints(RxDocumentServiceReques
         return this.getApplicableWriteEndpoints(request.requestContext.getExcludeRegions());
     }
 
-    public UnmodifiableList<URI> getApplicableWriteEndpoints(List<String> excludedRegions) {
+    public UnmodifiableList<URI> getApplicableWriteEndpoints(List<String> excludedRegionsOnRequest) {
         UnmodifiableList<URI> writeEndpoints = this.getWriteEndpoints();
 
-        if (excludedRegions == null ||
-            excludedRegions.isEmpty()) {
+        List<String> effectiveExcludedRegions = connectionPolicy.getExcludedRegions();
```

Comment:
kind feel the calculation of the effectiveRegions could belong to during request creation, since for diagnostics recording we might also need it.

Reply:
So, for the diagnostic string - both client-level and request-level `excludedRegions` are printed. It then becomes easier to determine the `effectiveExcludedRegions` since request-level setting overrides client-level setting anyway.